invenio_swh/tasks.py

```diff
@@ -83,16 +84,37 @@ def poll_deposit(self, id_):
     try:
         deposit = service.read(id_).deposit
         service.sync_status(deposit.id)
-        new_status = deposit.status
     except DepositFailed:
         # If the deposit already failed, we don't need to retry.
         return
 
     # Manually set status to FAILED on last retry.
     # Celery has a bug where it doesn't raise MaxRetriesExceededError, therefore we need to check retries manually.
     if self.request.retries == 5:
-        service.handle_status_update(deposit, SWHDepositStatus.FAILED)
+        service.update_status(deposit, SWHDepositStatus.FAILED)
         return
 
-    if new_status == SWHDepositStatus.WAITING:
+    if deposit.status == SWHDepositStatus.WAITING:
         raise DepositWaiting("Deposit is still waiting")
+
+
+@shared_task()
+def cleanup_depositions():
+    """Cleanup old depositions."""
+    if not current_app.config.get("SWH_ENABLED"):
+        current_app.logger.warning()
+        return
+    Deposit = service.record_cls.model_cls
+    # query for records that are stuck in "waiting" and were created in the previous 24 hours
+    res = Deposit.query.filter(
+        Deposit.status == SWHDepositStatus.WAITING,
+        Deposit.created >= datetime.now(timezone.utc) - timedelta(days=1),
```

Comment:
Would it make sense to query `Deposit.updated` instead? I imagine a case where we "manually" retry a deposit (e.g. from shell) and then it would get cancelled by this task (since it will end up in a `WAITING` status, and will have an old creation date).